src/Compilers/Core/Portable/InternalUtilities/Debug.cs

```diff
@@ -18,7 +18,13 @@ internal static class RoslynDebug
 
         /// <inheritdoc cref="Debug.Assert(bool, string)"/>
         [Conditional("DEBUG")]
-        public static void Assert([DoesNotReturnIf(false)] bool condition, string message)
+        public static void Assert([DoesNotReturnIf(false)] bool condition,
+#if NET
+            [CallerArgumentExpression(nameof(condition))] string? message = null
```

Comment:
>[CallerArgumentExpression(nameof(condition))] [](http://example.com/codeflow?start=12&length=45)

It is not clear to me how this is going to affect any existing usages of ```RoslynDebug.Assert```, given that the parameter was not optional and, therefore, all call sites supply its value explicitly.